### PR TITLE
feat(Image): imageRender、toolbarRender、onTransform、items、minScale、max…

### PR DESCRIPTION
**Pull request by linxianxi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/42845
https://github.com/ant-design/ant-design/issues/27273
https://github.com/ant-design/ant-design/issues/38418
https://github.com/react-component/image/issues/192
https://github.com/react-component/image/issues/110

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     support imageRender、toolbarRender、onTransform、items、minScale、maxScale. fix img common props are not passed to preview img     |
| 🇨🇳 Chinese |     支持 imageRender、toolbarRender、onTransform、items、minScale、maxScale。修复 img 的原生属性没有传递给预览图的问题     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 34cfae4</samp>

This pull request enhances the image component with new features, props, and demos, and updates the documentation and dependency accordingly. It adds the ability to customize the preview and toolbar render, to pass an array of preview sources to the preview group, and to render a video instead of an image in the preview. It also fixes some style and functionality issues of the preview feature, and some typos and inconsistencies in the documentation. It modifies the files `components/image/*`, and `package.json`.

### <samp>🤖 Generated by Copilot at 34cfae4</samp>

*  Add two new demos of custom toolbar render and custom preview render in the image component, and update the English and Chinese documentation with the links and descriptions of the new demos ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-831179c3c2c6ae10fc9ae3b08afc280a684665418994a76429025b0887aebcd5R1-R7), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-e0f4b6171a2d2ddfa3ab2e746e1fa26aacb7cf8ec6a46cbc3d2250f75507b0e6R1-R24), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-ff24c471c0b5319a02b8f90d81007f652989af18a8555822ca98d324c8b00d73R1-R7), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-94187e7b726a3e17e6c68962179363e372b1b38b03c72eb8d78cc5cf2bff3852R1-R56), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-6a52026f5728b46e5b93ce0149ddfd4b94f79896c72f98ad4cbfa9ed2843a8b2R27-R28), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-f5b985fbb4265b0e83d262d5f201a0b268c58dcadfde017e1126a4c1e6c15281R28-R29))
*  Replace the original demo of preview group visible in the image component with a new one that uses the new `items` prop to pass an array of preview sources to the `Image.PreviewGroup` component ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-9bcd361df3114c2fcb93b12a44d96953f03442fc9206eefba5cf6c53a0972ab8L1-R18))
*  Add a heading for the `Image` component API in the English and Chinese documentation ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-6a52026f5728b46e5b93ce0149ddfd4b94f79896c72f98ad4cbfa9ed2843a8b2R35-R36), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-f5b985fbb4265b0e83d262d5f201a0b268c58dcadfde017e1126a4c1e6c15281R36-R37))
*  Update the type of the `preview` prop in the `Image` component API from `previewType` to `PreviewType` in the English and Chinese documentation, to match the naming convention of the other types ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-6a52026f5728b46e5b93ce0149ddfd4b94f79896c72f98ad4cbfa9ed2843a8b2L39-R50), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-f5b985fbb4265b0e83d262d5f201a0b268c58dcadfde017e1126a4c1e6c15281L40-R44))
*  Restructure the `previewType` section in the English and Chinese documentation, and add the detailed descriptions and versions for each prop in the `PreviewType`, `PreviewGroupType`, `TransformType`, `TransformAction`, and `ToolbarRenderType` interfaces ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-6a52026f5728b46e5b93ce0149ddfd4b94f79896c72f98ad4cbfa9ed2843a8b2L63-R178), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-f5b985fbb4265b0e83d262d5f201a0b268c58dcadfde017e1126a4c1e6c15281L46-R161))
*  Add the new props `imageRender`, `onTransform`, `minScale`, and `maxScale` to the `PreviewType` and `PreviewGroupType` interfaces, and the new prop `current` to the `imageRender` parameter in the `PreviewType` interface, to support custom preview render and transform events in the image component ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-6a52026f5728b46e5b93ce0149ddfd4b94f79896c72f98ad4cbfa9ed2843a8b2L63-R178), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-f5b985fbb4265b0e83d262d5f201a0b268c58dcadfde017e1126a4c1e6c15281L46-R161))
*  Update the style of the preview operations container and icons in the image component, and reverse the order of the icons to align them to the end of the flex container ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-922f1030835ae207735e4fda28ea766dc4a4e35ae7ba47db9d084a4c0b2fec3eL67-R67), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-922f1030835ae207735e4fda28ea766dc4a4e35ae7ba47db9d084a4c0b2fec3eL90-R90))
*  Add the style of the preview content container in the image component, and center the content in the flex container ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-922f1030835ae207735e4fda28ea766dc4a4e35ae7ba47db9d084a4c0b2fec3eR185-R187))
*  Update the style of the preview content wrapper and content in the image component, and remove the `pointerEvents` and `transition` properties from the wrapper, and move the `pointerEvents` property to the content itself, to avoid blocking the mouse events on the custom preview content ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-922f1030835ae207735e4fda28ea766dc4a4e35ae7ba47db9d084a4c0b2fec3eL195-R200), [link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-922f1030835ae207735e4fda28ea766dc4a4e35ae7ba47db9d084a4c0b2fec3eL221-L224))
*  Update the dependency version of the `rc-image` package from `5.17.1` to `5.18.1`, to use the latest features and bug fixes of the image component ([link](https://github.com/ant-design/ant-design/pull/43075/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L132-R132))

### CODE REVIEW

File: components/image/demo/imageRender.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const App: React.FC = () => (
+  <Image
+    width={200}
+    preview={{
+      imageRender: () => (
+        <video
+          muted
+          width="100%"
+          controls
+          style={{ pointerEvents: 'auto' }}
```
Comment (linxianxi) [inline components/image/demo/imageRender.tsx:13]: imageRender 要加 pointerEvents: 'auto'，感觉这样不太好

**Comment by linxianxi** [comment]
@afc163

File: components/image/demo/imageRender.tsx (hunk 1 above)
Comment (zombieJ) [reply components/image/demo/imageRender.tsx:13]: 这个是为啥？

File: components/image/index.en-US.md [hunk 2]
```diff
@@ -24,43 +24,139 @@ Previewable image.
...
+| getContainer | The mounted node for preview dialog but still display at fullScreen | string \| HTMLElement \| (() => HTMLElement) \| false | - | 4.8.0 |
+| mask | Thumbnail mask | ReactNode | - | 4.9.0 |
+| maskClassName | The className of the mask | string | - | 4.11.0 |
+| rootClassName | The classname of the preview root DOM | string | - | 5.4.0 |
+| scaleStep | `1 + scaleStep` is the step to increase or decrease the scale | number | 0.5 | - |
+| minScale | Min scale | number | 1 | 5.7.0 |
+| maxScale | Max scale | number | 50 | 5.7.0 |
+| forceRender | Force render preview dialog | boolean | - | - |
+| toolbarRender | Custom toolbar render | (params: Omit<[ToolbarRenderType](#ToolbarRenderType), 'current' \| 'total'>) => React.ReactNode | - | 5.7.0 |
+| imageRender | Custom preview content | { originalNode: React.ReactNode, transform: [TransformType](#TransformType) } => React.ReactNode | - | 5.7.0 |
```
Comment (zombieJ) [inline components/image/index.en-US.md:66]: 刚刚开始的时候文档别透出太多属性，以后收不回来。一般 antd xxxRender 的风格应该是 `xxxRender(oriNode, info)`，不是直接的 `xxxRender(info)`

Comment (linxianxi) [reply components/image/index.en-US.md:66]: 这个是偏右佬叫我放一个对象里的（方便扩展），我本来是你说的这个风格的

Comment (zombieJ) [reply components/image/index.en-US.md:66]: ref https://github.com/ant-design/ant-design/wiki/API-Naming-rules

Comment (zombieJ) [reply components/image/index.en-US.md:66]: 然后这边 `info` 可以先不给

File: components/image/index.en-US.md [hunk 3]
```diff
@@ -24,43 +24,139 @@ Previewable image.
...
+| onTransform | Callback when the transform of image changed | { transform: [TransformType](#TransformType), action: [TransformAction](#TransformAction) } | - | 5.7.0 |
```
Comment (zombieJ) [inline components/image/index.en-US.md:67]: 这个如果没有具体需求可以也先不给

File: components/image/demo/imageRender.tsx (hunk 1 above)
Comment (linxianxi) [reply components/image/demo/imageRender.tsx:13]: 因为预览里的 img 本身是有这个属性的，如果自定义 render 的话也是要加的。像 Modal 组件其实外面也是加了 pointerEvents: auto。都是用的 rc-dialog，是不是实现要改一下？

File: components/image/index.en-US.md (hunk 3 above)
Comment (linxianxi) [reply components/image/index.en-US.md:67]: 这个有 issue，说是要旋转后的回调

File: components/image/index.en-US.md (hunk 2 above)
Comment (linxianxi) [reply components/image/index.en-US.md:66]: https://github.com/react-component/image/pull/248#discussion_r1221156145  要改的话应该来得及？

**Review by zombieJ — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Pull request merged**